### DISCUSSION TITLE
1.0 STABLE RELEASE OADMAP

### DISCUSSION THREAD

**Discussion by k11q** [unovue/reka-ui#230] [General]
I dont want to call it beta, because it doesnt mean anything in our case. A stable release means no API/breaking changes, and every main features including keyboard, screen reader, mouse/pointer accessibility are covered, styling with related states are covered and all API are implemented and working. A stable release is where the library is reliable enough to be used for production. Below lies our roadmap for the stable release.

## Accordion
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Alert Dialog
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Aspect Ratio
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Avatar
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Checkbox
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Collapsible
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Context Menu
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Dialog
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Dropdown Menu
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Hover Card
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Label
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Menubar
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Navigation Menu
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Popover
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Progress
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Radio Group
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Scroll Area
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Select
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Separator
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Slider
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Switch
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Tabs
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Toggle
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Toggle Group
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Toolbar
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

## Tooltip
- [x] Anatomy implementation
- [ ] API implementation and finalised
- [ ] Keyboard Accessibility
- [ ] Mouse/Pointer Accessibility
- [ ] Role, Property, State, and Tabindex Attributes
- [ ] Screen reader accessibility (The screen reader recording should match EXACTLY like radix's)

**Comment by Blinks44** [comment]
What happened with Toast component? It won’t be implemented?

**Reply by k11q** [reply]
hi yeah we think it is better to use something like [vue-sonner](https://github.com/xiaoluoboding/vue-sonner) for it which is a really great option and not waste time reinventing the wheel

**Reply by ojvribeiro** [reply]
Now Vue Sonner was a nice discovery ♥ Thanks!

**Comment by 9mm** [comment]
Interesting, so you already have -all- the radix-ui core elements in (even if theyre not fully accessible yet, etc). Am i reading this right? If so, that's awesome

**Reply by zernonia** [reply]
Hey @9mm . Yup we have almost all the components from RadixUI. I've tested the accessbility (with aXe) and it should be all good now.

We still have to write test on some components, however we are excited for public to use and test it out.
Currently it is being use to develop https://github.com/radix-vue/shadcn-vue, and it has been great thus far 😁

**Reply by 9mm** [reply]
That is incredible. And... I realize no one can say for certain on anything, but how do you feel about the long term life of this project? I'm about to start a new project that is actually a bit serious and I really want to use this, as one area of vue that is just downright abysmal are the UI libraries are immensely weak. I really love radix (and shadcn). It's not SO serious that it would be the end of the world to redo components if something happened to this project, but also definitely not something id want to do.
